### PR TITLE
MMT-3526: Implements the Provider Permissions form for groups.

### PR DESCRIPTION
**Pull request by abbottry**
# Overview

### What is the feature?

Adds the permissions form for Provider lever permissions that apply to a group.

### What is the Solution?

Added the checkbox table / form that _shows_ the permissions and which permissions are selected for a given group.

### What areas of the application does this impact?

Group Provider level permissions.

# Testing

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/App.jsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ import useNotificationsContext from './hooks/useNotificationsContext'
 import { GET_ACLS } from './operations/queries/getAcls'
 
 import withProviders from './providers/withProviders/withProviders'
+import ProviderPermissionsPage from './pages/ProviderPermissionsPage/ProviderPermissionsPage'
```
Comment (macrouch) [inline static/src/js/App.jsx:43]: Move import to other pages imports

File: static/src/js/components/ProviderPermissions/ProviderPermissions.jsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import React, { useCallback } from 'react'
+
+import { Row } from 'react-bootstrap'
```
Comment (macrouch) [inline static/src/js/components/ProviderPermissions/ProviderPermissions.jsx:3]: Wrong import for bootstrap

File: static/src/js/components/ProviderPermissions/ProviderPermissions.jsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+  })
+
+  /**
+   * Determine if a group has a permission for a target
+   * @param {String} target The target identity of the Acl
+   * @param {String} permission One of create, read, update or delete
+   * @returns Whether or not the group has the permission
+   */
+  const checkAclPermission = (target, permission) => {
```
Comment (macrouch) [inline static/src/js/components/ProviderPermissions/ProviderPermissions.jsx:57]: Should this be a util function outside the component?

File: static/src/js/components/ProviderPermissions/__tests__/ProviderPermissions.test.jsx [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+import { render, screen } from '@testing-library/react'
+import { Cookies } from 'react-cookie'
+import { MockedProvider } from '@apollo/client/testing'
+import {
+  MemoryRouter,
+  Route,
+  Routes
+} from 'react-router'
+import userEvent from '@testing-library/user-event'
+import { describe } from 'vitest'
```
Comment (macrouch) [inline static/src/js/components/ProviderPermissions/__tests__/ProviderPermissions.test.jsx:11]: This import shouldn't be needed

File: static/src/js/pages/ProviderPermissionsPage/__tests__/ProviderPermissionPage.test.jsx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+import { render, screen } from '@testing-library/react'
+import { Cookies } from 'react-cookie'
+import { MockedProvider } from '@apollo/client/testing'
+import {
+  MemoryRouter,
+  Route,
+  Routes
+} from 'react-router'
+import userEvent from '@testing-library/user-event'
+import { describe } from 'vitest'
```
Comment (macrouch) [inline static/src/js/pages/ProviderPermissionsPage/__tests__/ProviderPermissionPage.test.jsx:11]: This import shouldn't be needed

**Review by trevorlang — APPROVED** [review]

**Review by macrouch — APPROVED** [review]

**Pull request merged**